portfolio/src/main/webapp/script.js

```diff
@@ -14,15 +14,38 @@
 
 /**
  * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+*/
+
+const KEY = "&key=AIzaSyAAHC0kzUB8IDwJlG0DaP2lLyc_haNkNWs"
+const URL = "https://maps.googleapis.com/maps/api/geocode/json?latlng="
+
+async function generateQuery(longitude,latitude){ // function wont have parameters this is just for offline testing.
+
+	//let longitude = document.getElementById("lon").value // from the dummy html made for testing purpouses
+	//let latitude = document.getElementById("lat").value
+
+    let geoCodeRequest = URL + latitude + "," + longitude + KEY // creating the search Key to get human readable location
+	let location = ""
+    console.log(geoCodeRequest) //Incase we want to look at the entire JSON in browser
+
+	await fetch(geoCodeRequest).then(response => response.json())
+	.then(data => {
+		
+			location = data.results[0].formatted_address; // get the common name of the coordinates
+		
+		})
```

Comment:
Since you're using `await` here you can bypass the Promise interface and use the returned result directly without having to rely on `.then()`:
```suggestion
let response = fetch(geoCodeRequest);
let data = response.json();
location = data.results[0].formatted_address;

```